src/hypothesis/internal/conjecture/engine.py

```diff
@@ -1541,6 +1543,60 @@ def reoffset(o):
 
         minimize_int(offset, reoffset)
 
+    def shrink_offset_pairs(self):
+        """Lower any two blocks offset from each other the same ammount.
+
+        Try lowering every non-zero pair of non-shrinking blocks by a
+        common offset.
+
+        """
+        self.debug('Shrinking offset pairs.')
+
+        current = self.shrink_target
+        blocked = [current.buffer[u:v] for u, v in current.blocks]
+
+        n = len(blocked)
+
+        shrink_blocks = [b for b in range(n) if self.is_shrinking_block(b)]
+
+        ints = [int_from_bytes(blocked[i]) for i in range(n)]
+
+        nonzero = sorted([b for b in range(n)
+                          if ints[b] > 0 and b not in shrink_blocks])
+
+        if len(nonzero) == 0:
+            return
+
+        # TODO: Should we try combinations of > 2?
+        pairs = itertools.combinations(nonzero, 2)
```

Comment:
As @Zac-HD pointed out in his comment, we've got a slightly weird style of loop in the shrinker and cases like this are why we need it: If we make changes to the current `shrink_target` then the set of blocks could change fairly drastically, so we need to write loops which are robust against those changes.

You could for example do something like:

```python    
i = 0
while i < len(self.blocks):
  if not self.is_shrinking_block(i):
    j = i + 1
    while j < len(self.shrink_target.blocks):
      if not self.is_shrinking_block(j):
        # TODO: Shrink logic for pair (i, j) goes here.
      j += 1
  i += 1
```